app/models/katello/glue/candlepin/product.rb

```diff
@@ -32,14 +32,22 @@ def self.validate_name(name)
       name.gsub(/[^a-z0-9\-_ ]/i, "")
     end
 
-    def self.import_from_cp(attrs = nil, &block)
+    def self.engineering_product_id?(id)
+      id.to_i.to_s == id
```

Comment:
This seems hacky (I know you just copied it from below). Why not use regex? It's meant for this kind of format filtering.

i.e. `id.match? /^\d+$/`

Reply:
It is really hacky, but its hacky in candlepin and we're just dealing with it.  I'm fine using a regex, i usually shy away from them but this is simple enough and likely will not 'evolve' in complexity.